activerecord/test/cases/adapters/abstract_mysql_adapter/connection_test.rb

```diff
@@ -216,6 +216,51 @@ def test_release_non_existent_advisory_lock
       "expected release_advisory_lock to return false when there was no lock to release"
   end
 
+  def test_version_string
+    @connection.stub(:get_full_version, "8.0.35-0ubuntu0.22.04.1") do
+      assert_equal "8.0.35", @connection.get_database_version.to_s
+    end
+
+    @connection.stub(:get_full_version, "5.7.0") do
+      assert_equal "5.7.0", @connection.get_database_version.to_s
+    end
+  ensure
+    @connection.pool.pool_config.server_version = nil
```

Comment:
I don't understand why there's an `ensure` block setting `server_version` to `nil`. Calling `get_database_version` doesn't set `server_version` on the pool config. Before and after this is called `@connection.pool.server_version(@connection)` returns `#<ActiveRecord::ConnectionAdapters::AbstractAdapter::Version:0x0000000105bf4b30 @version=[8, 0, 36], @full_version_string="8.0.36">`. So changing it to `nil` is would be unsetting the version from what it was when the test starts.